app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -0,0 +1,197 @@
+import androidx.compose.runtime.*
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavDestination
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.currentBackStackEntryAsState
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun MainScreen() {
+    val navController = rememberNavController()
+    val navBackStackEntry by navController.currentBackStackEntryAsState()
+    var showBottomBar by remember { mutableStateOf(true) }
+    var showBackIcon = !showBottomBar
+    Scaffold(
+        topBar = {
+            TopBar(
+                navigationIcon = {
+                    if (showBackIcon) {
+                        navController.navigateUp()
+                        BackIcon(onBackClick = navController::popBackStack)
+                    }
+                }
+            )
+        },
+        bottomBar = {
+            if (showBottomBar) {
+                BottomNavigationBar(
+                    destinations = listOf(
+                        NavigationItem.HomeDestination,
+                        NavigationItem.FavoritesDestination,
+                    ),
+                    onNavigateToDestination = {
+                        navController.navigate(it.route) {
+                            popUpTo(it.route) {
+                                inclusive = true
+                            }
+                        }
+                    },
+                    currentDestination = navBackStackEntry?.destination
+                )
+            }
+        }
+    ) { padding ->
+        Surface(
+            color = MaterialTheme.colors.background,
+            modifier = Modifier.fillMaxSize()
+        ) {
+            NavHost(
+                navController = navController,
+                startDestination = NavigationItem.HomeDestination.route,
+                modifier = Modifier.padding(padding)
+            ) {
+                composable(NavigationItem.HomeDestination.route) {
+                    showBottomBar = true
+                    showBackIcon = showBackIcon
+                    HomeRoute(
+                        onNavigateToMovieDetails = { movieId ->
+                            navController.navigate(
+                                NavigationItem.MovieDetailsDestination.createNavigationRoute(movieId)
+                            )
+                            showBottomBar = false
+                        }
+                    )
+                }
+                composable(NavigationItem.FavoritesDestination.route) {
+                    showBottomBar = true
```

Comment:
This solution doesn’t scale well. Imagine that you need to add more screens, than you’ll have to put the logic for showing bottomBar/backIcon into every screen:
```kotlin
composable(someRoute) {
  showBottomBar = true/false
  showBackIcon = false/true
}
```
You can use provided navBackStackEntry for calculating showBottomBar:
```kotlin
val showBottomBar = when (navBackStackEntry?.destination?.route) {
    MovieDetailsDestination.route -> false
    else -> true
}

val showBackIcon = !showBottomBar
```